chain/index/chain-indexing-overview-for-rpc-providers.MD

```diff
@@ -0,0 +1,202 @@
+  EnableIndexer = true 
+
+[Fevm]
+# Enable the ETH RPC APIs.
+  EnableEthRPC = true
+
+[Events]
+# Enable the Actor Events APIs.
+  EnableActorEventsAPI = true
+```
+
+You can learn more about these configuration options and other configuration options available for the `ChainIndexer` [here](https://github.com/filecoin-project/lotus/blob/master/documentation/en/default-lotus-config.toml).
+
+
+### Garbage Collection
+
+The `ChainIndexer` includes a garbage collection (GC) mechanism to manage the amount of historical data retained. By default, GC is disabled to preserve all indexed data.
+
+To configure GC, use the `GCRetentionEpochs` parameter in the `ChainIndexer` section of your config.
+
+The ChainIndexer [periodically runs](https://github.com/filecoin-project/lotus/blob/master/chain/index/gc.go#L15) GC if `GCRetentionEpochs` is > 0 and removes indexed data for epochs older than `(current_head_height - GCRetentionEpochs)`.
+
+```toml
+[ChainIndexer]
+  GCRetentionEpochs = X  # Replace X with your desired value
+```
+
+- Setting `GCRetentionEpochs` to 0 (**default**) disables GC.
+- Any positive value enables GC and determines the number of epochs of historical data to retain.
+
+#### Recommendations
+
+1. **Archival Nodes**: **Keep GC disabled** (`GCRetentionEpochs` = 0) to retain all indexed data.
+
+2. **Non-Archival Nodes**:  Set `GCRetentionEpochs` to match the amount of chain state your node retains 
+(*Example:* if your node is configured to retain 2 days of Filecoin chain state with the Splitstore, set `GCRetentionEpochs` to `retentionDays * epochsPerDay = 2 * 2880 = 5760`).
+**Warning:** Setting this value below the chain state retention period will degrade RPC performance and reliability because the Index will lack data for epochs still present in the chain state.  
+
+### Removed Options
+
+**Note: The following config options no longer exist in Lotus and have been removed in favor of the ChainIndexer config options explained above. They can be removed when upgrading to Lotus v1.31.0.**
+
+```toml
+[Fevm]
+EthTxHashMappingLifetimeDays = 0
+
+[Events]
+DisableHistoricFilterAPI = false
+DatabasePath = ""
+```
+
+## Migration Guide
+
+Migrating to the new `ChainIndexer` involves several steps to ensure a smooth transition:
+
+1. **Stop the Lotus Node**
+   - Stop your Lotus node before starting the upgrade and migration process.
+   - 
+2. **Update Configuration**
+   - Modify your Lotus configuration to enable the `ChainIndexer` as described in the [`ChainIndexer Config` section above](#chainindexer-config).
+
+3. **Restart Lotus Node**
+   - Restart your Lotus node with the new configuration.
+   - The `ChainIndexer` will begin indexing **real-time chain state changes** immediately in the `${LOTUS_PATH}/chainindex` directory.
+   - **However, it will not automatically index any historical chain state (i.e., any previously existing chain state). To perform backfilling, please see the [`Backfilling` section below](#backfilling).**
+
```

Comment:
```suggestion

Note: The ChainIndexer will write state to `${LOTUS_HOME}/chainindex` rather than the [removed option](#removed-options) for `Events.DatabasePath` (which defaulted to `${LOTUS_HOME}/sqllite`).  It's recommended to keep the legacy directory around until you've confirmed you don't need to [rollback](#rollback).
```

I'm thinking to be clear about the sqllite directory and when you can get rid of it.